spec/decorators_spec.rb

```diff
@@ -0,0 +1,23 @@
+require './capitalized_decorator'
+require './trimmer_decorator'
+require './person'
+
+describe 'Decorators' do
+  it 'creates a new base decorator object' do
+    nameable = 'maximilianus'
+    base_decorator = BaseDecorator.new(nameable)
+    expect(base_decorator).to be_an_instance_of(BaseDecorator)
+  end
+
+  it 'should capatlize an inserted name' do
+    person = Person.new(777, 23, 'maximilianus')
+    capatlize = CapitalizedDecorator.new(person)
+    expect(capatlize.correct_name).to eq('Maximilianus')
+  end
+
+  it 'Should trim names more than 10 charceters' do
+    person = Person.new(777, 23, 'maximilianus')
+    trimmer = TrimmerDecorator.new(person)
+    expect(trimmer.correct_name).to eq('maximilian')
+  end
+end
```

Comment:
- [ ] Good job for testing the Decorator classes.🥇 Please make sure to test the method, `correct_name`,  present in the `BaseDecorator` class 📌
- [ ] I've noticed that currently you are missing the tests of the `Nameable` class, kindly include it so that all the methods in all the classes are tests as mentioned in the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md) ✨ For example, you can try something like below 👇


```ruby
describe 'Nameable' do
  let(:nameable) { Nameable.new }

  it 'raises NotImplementedError when correct_name is not implemented' do
    expect { nameable.correct_name }.to raise_error(NotImplementedError)
  end
end
```